pyproject.toml

```
@@ -38,7 +38,7 @@ description = ""
 authors = ["Hiroshiba <hihokaruta@gmail.com>"]
 
 [tool.poetry.dependencies]
-python = "~3.8,>=3.8.1"
+python = "~3.11"
```

build.ymlなどと揃えるために、バージョン固定で`3.11.3`にしちゃうのどうでしょう。
（ほんとは他のとこもバージョン固定したほうが良さそうですが、まあおいおい･･･！）
```suggestion
python = "3.11.3"
```

---

2か月に1度、新しいPythonのリビジョンが出るたびにPRを出して、リポジトリのPythonを更新することを考えていますが、そのたびにローカルのPython環境を切り替える必要が出ますね。2か月に1度なら許容できますかね？

個人的には固定してしまっても大丈夫ですが、認識の確認だけしたいです。

---

なるほどです。パッチバージョンはここでは固定しないのが良いのかなと思いました！

ただ、開発環境とリリース環境でPythonのパッチバージョンが異なるのはもしかしたら何か問題になるかもですね。
一回問題が起きたら考えるのが良いかなぁ。

ちなみにリビジョンのたびに更新する理由は、やはりセキュリティ上の懸念でしょうか。

---

> 開発環境とリリース環境でPythonのパッチバージョンが異なるのはもしかしたら何か問題になるかも

その可能性はありますね。起きてみないとわからない...

> ちなみにリビジョンのたびに更新する理由は、やはりセキュリティ上の懸念でしょうか。

そうですね。
バグ修正・セキュリティ修正が入ってくるので、CI側は積極的に更新できるようにしておきたい、くらいの気持ちです！

masterブランチで更新しても、リリースに反映するのは機能リリース（VOICEVOXのマイナーバージョン更新）のタイミングだと思うので、そのタイミングでリビジョンが最新になっているといいな、というくらいです。

`pyproject.toml`のPythonバージョンでは、問題になるようなバグが見つかることがあれば、別途修正バージョン以上・影響のないバージョンに固定する、という想定で、PR時点ではマイナーバージョンの固定にしていました。

---

なるほどです！！
個人的には、パッチバージョン違いによるエラー発生よりも、リリース時のパッチバージョンが低いことによる（セキュリティなどの）問題抑制のが大事かなと思いました。

ので（少なくともエンジンは）言語のマイナーバージョンまで固定で良いのかなと思いました！！

@y-chan 共有です🙏